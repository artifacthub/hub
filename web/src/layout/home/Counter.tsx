import isNull from 'lodash/isNull';
import React from 'react';

import styles from './Counter.module.css';

interface Props {
  name: string;
  isLoading: boolean;
  value: number | null;
}

const Counter = (props: Props) => (
  <div className={`text-center ${styles.counterWrapper}`}>
    {props.isLoading ? (
      <div className={`h3 ${styles.counter}`}>
        <div role="status" className="spinner-grow text-primary" />
      </div>
    ) : (
      <div className={`h3 ${styles.counter}`}>{isNull(props.value) || props.value === 0 ? '-' : props.value}</div>
    )}
    <small className="text-uppercase">{props.name}</small>
  </div>
);

export default Counter;
