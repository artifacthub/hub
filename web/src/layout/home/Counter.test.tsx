import { render } from '@testing-library/react';
import React from 'react';

import Counter from './Counter';

const defaultProps = {
  name: 'name',
  isLoading: false,
  value: 12,
};

describe('Counter', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Counter {...defaultProps} />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders proper content', () => {
    const { getByText } = render(<Counter {...defaultProps} />);
    expect(getByText(defaultProps.value!.toString())).toBeInTheDocument();
    expect(getByText(defaultProps.name)).toBeInTheDocument();
  });

  it('renders loading', () => {
    const { getByRole, getByText } = render(<Counter {...defaultProps} isLoading />);
    expect(getByRole('status')).toBeInTheDocument();
    expect(getByText(defaultProps.name)).toBeInTheDocument();
  });

  it('renders placeholder if value is 0 or null', () => {
    const { getByText } = render(<Counter {...defaultProps} value={0} />);
    expect(getByText('-')).toBeInTheDocument();
  });
});
