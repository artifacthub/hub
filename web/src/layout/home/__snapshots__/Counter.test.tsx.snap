// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `[Function]`;
